### ISSUE TITLE
"SAME" padding for avg_pool2d yields different results than explicitly 0-padded input with "VALID"

### ISSUE THREAD

**Issue by Lescurel** [tensorflow/tensorflow#45579]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- TensorFlow installed from (source or binary): binary 
- TensorFlow version (use command below): v2.3.0-rc2-23-gb36436b087 2.3.0
- Python version: 3.8.5

**Describe the current behavior**

When using padding "SAME" with average pooling, the results on a small input look wrong. If I use average pooling on a completely empty matrix with only 1 value of the matrix at  1, using the "SAME" padding algorithm I don't get a correct average value. If I explicitly 0-pad the input and use "VALID", I get the correct value. 

**Describe the expected behavior**

The behavior should be the same using :
* an explicitly 0-padded matrix with `tf.pad` and the "VALID" algorithm
* The "SAME" padding algorithm

**Standalone code to reproduce the issue**

**"SAME"**
```python
import tensorflow as tf
import numpy as np
generated = np.zeros((4,4))
generated[2,1] = 1
gen_nhwc = tf.constant(generated[np.newaxis,:,:,np.newaxis])
pool = 3
res = tf.nn.avg_pool2d(gen_nhwc, (pool,pool), (1,1),"SAME")
result = np.squeeze(res.numpy())
print("Input:\n",generated)
print("Output\n",result)
```
```
Input:
 [[0. 0. 0. 0.]
 [0. 0. 0. 0.]
 [0. 1. 0. 0.]
 [0. 0. 0. 0.]]
Output
 [[0.         0.         0.         0.        ]
 [0.16666667 0.11111111 0.11111111 0.        ]
 [0.16666667 0.11111111 0.11111111 0.        ]
 [0.25       0.16666667 0.16666667 0.        ]]
```

**`tf.pad` and "VALID" results:** 
```python
import tensorflow as tf
import numpy as np
generated = np.zeros((4,4))
generated[2,1] = 1
gen_nhwc = tf.constant(generated[np.newaxis,:,:,np.newaxis])
pool = 3
paddings = [[0,0],[pool//2,pool//2],[pool//2,pool//2],[0,0]]
gen_pad = tf.pad(gen_nhwc, paddings, "CONSTANT")
res = tf.nn.avg_pool2d(gen_pad, (pool,pool), (1,1),"VALID")
result = np.squeeze(res.numpy())
print(generated)
print(result)
```

```
Input:
 [[0. 0. 0. 0.]
 [0. 0. 0. 0.]
 [0. 1. 0. 0.]
 [0. 0. 0. 0.]]
Output
 [[0.         0.         0.         0.        ]
 [0.11111111 0.11111111 0.11111111 0.        ]
 [0.11111111 0.11111111 0.11111111 0.        ]
 [0.11111111 0.11111111 0.11111111 0.        ]]
```

**Comment by ravikyram** [comment]
I have tried in colab with TF 2.2, 2.3 and nightly version(`2.5.0-dev20201210`) and was able to reproduce the issue. Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/d100d4b85685c6ccf0ac70182aeab3e9/untitled574.ipynb).Thanks!

**Comment by Rulino** [comment]
Thanks for creating issue, thought about doing so as well according to my yesterdays question on SO ([Question](https://stackoverflow.com/questions/65231630/understanding-average-sum-pooling-padding-in-keras/65237472)). This happened to me in colab with TF 2.3 as well as at windows 10, python 3.6 with TF 2.2 and 1.14, seem to be a bug across all versions, can somebody confirm?

**Comment by saikumarchalla** [comment]
Was able to Reproduce the issue in TF 2.5 as well. Please find the gist [here](https://colab.research.google.com/gist/saikumarchalla/bccaa47ea61ecd98ce5dfa6d15e4d6ed/untitled88.ipynb).Thanks!

**Comment by tilakrayal** [comment]
@Lescurel,
I tried to execute the code in the alternative approach and it was executed without any issue/error & also the output was also the expected and similar to the initial code. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/a3077a7abaeee3a9708edecc2ec8c11d/untitled1097.ipynb). Thank you!
